Free the m_inherited bit in StylePropertyMetadata by converting clients to using isInheritedProperty() function

weinig opened the pull request:
#### 8f1522c3ad10b76dd549119dee88c87f9b267888
<pre>
Free the m_inherited bit in StylePropertyMetadata by converting clients to using isInheritedProperty() function
<a href="https://bugs.webkit.org/show_bug.cgi?id=287109">https://bugs.webkit.org/show_bug.cgi?id=287109</a>

Reviewed by Antti Koivisto and Darin Adler.

Free up a bit on StylePropertyMetadata by switching single user of the bit
to use the lookup function `isInheritedProperty()` instead.

* Source/WebCore/css/CSSProperty.h:
* Source/WebCore/css/StyleProperties.h:
* Source/WebCore/style/PropertyCascade.cpp:

Canonical link: <a href="https://commits.webkit.org/289942@main">https://commits.webkit.org/289942@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a05cabc0a6763cc92e5437c80762c4c0dff2136b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/88351 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/7870 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/42777 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/93308 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/39105 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/90402 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/8257 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/16053 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/68146 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/25868 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/91353 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/6310 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/79910 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/48515 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/6082 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/34360 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/38213 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/76464 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/35205 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/95151 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/15526 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/11378 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/77018 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/15782 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/75763 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/76268 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/18799 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/20654 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/19074 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/8563 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/15542 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/20849 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/15283 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/18732 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/17065 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/style/PropertyCascade.cpp:
```diff
@@ -248,11 +248,12 @@ bool PropertyCascade::addMatch(const MatchedProperties& matchedProperties, Casca
             }
 
-            if (m_includedProperties.contains(PropertyType::Inherited) && current.isInherited())
+            bool currentIsInherited = CSSProperty::isInheritedProperty(current.id());
```
If this is an expensive operation we should make sure to only do it if the result is needed. I can't tell from the structure of the code if it is always used.

weinig replied:
I couldn't tell either. Is it common for m_includedProperties to include both PropertyType::Inherited and PropertyType::NonInherited? @anttijk do you know?

anttijk replied:
That's the common case. 
```
static constexpr OptionSet<PropertyType> normalProperties() { return { PropertyType::NonInherited,  PropertyType::Inherited }; }
```

anttijk replied:
I would be surprised if this has any perf significance (and if it did we should be using the cached bit in StyleBuilder.cpp too).

darinadler reviewed (approved):
A surprising performance cost would be the only risk here.

weinig committed:
Free the m_inherited bit in StylePropertyMetadata by converting clients to using isInheritedProperty() function
https://bugs.webkit.org/show_bug.cgi?id=287109

Reviewed by Antti Koivisto and Darin Adler.

Free up a bit on StylePropertyMetadata by switching single user of the bit
to use the lookup function `isInheritedProperty()` instead.

* Source/WebCore/css/CSSProperty.h:
* Source/WebCore/css/StyleProperties.h:
* Source/WebCore/style/PropertyCascade.cpp:

Canonical link: https://commits.webkit.org/289942@main
